Automattic/jetpack
# CRM: Adding Automation contact triggers

## Proposed changes:

* This PR adds Automation Contact triggers. The files with triggers will likely need changes as we better understand what data is being passed in and out, but currently all triggers take contact objects (with modified data depending on what was passed into the object, such as in this example: https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Contacts.php#L3007)
* This PR also includes tests

### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/3097


## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Run tests by navigating to the CRM directory in your local installation after running `jetpack build plugins/crm`, and then run `composer phpunit -- --testdox --testsuite automation` (note this will currently run all tests as well with a new commit adding tests to this PR).
* To test the objects that would be passed to the actions within each trigger, you can add an `error_log` within the relevant hook such as `zeroBSCRM_IA_EditCustomerWPHook` (within each case), such as `error_log('Firing the contact update: ' . json_encode( $obj ) );`. 
* You can then test updating (status / email / any contact field) / deleting / adding a new contact, and then make sure the $obj from the error log you are expecting is fired.
